### ISSUE TITLE
Template engine not renaming primaryOutputs paths

### ISSUE THREAD

**Issue by ccpu** [dotnet/templating#2682]
This issue has been fixed in (#1566), but I can't get it to work, here is my template.json:

```
{
  "$schema": "http://json.schemastore.org/template",
  "author": "Me",
  "classifications": [
    "Common",
    "Console",
    "C#8"
  ],
  "identity": "ExampleTemplate.AsyncProject",
  "name": "Example templates: async project",
  "shortName": "new-template",
  "sourceName": "$safeProjectName$",
  "tags": {
    "language": "C#",
    "type": "project"
  },
  "primaryOutputs": [
    {
      "path": "/$safeProjectName$/$safeProjectName$.csproj"
    }
  ],
  "postActions": [
    {
      "description": "Restore NuGet packages required by this project.",
      "manualInstructions": [
        {
          "text": "Run 'dotnet restore'"
        }
      ],
      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
      "continueOnError": true
    }
  ]
}
```

when I am running `dotnet new new-template -n NewProject` will receive following error:

```
Processing post-creation actions...
Running 'dotnet restore' on /$safeProjectName$/$safeProjectName$.csproj...
MSBUILD : error MSB1001: Unknown switch.
Switch: /$safeProjectName$/$safeProjectName$.csproj

For switch syntax, type "MSBuild -help"

Restore failed.
Post action failed.
Description: Restore NuGet packages required by this project.
Manual instructions: Run 'dotnet restore'
```
That's because `$safeProjectName$` not renamed to `NewProject`.

`dotnet new --debug:version` result:

```
Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.404
```

Repo:

[https://github.com/ccpu/dotnet-template-primaryOutputs-issue](https://github.com/ccpu/dotnet-template-primaryOutputs-issue)

**Comment by DavidKarlas** [comment]
I think problem is that primaryOutput thats with `/` :
https://github.com/ccpu/dotnet-template-primaryOutputs-issue/blob/main/.template.config/template.json#L19
should be:
```
"path": "$safeProjectName$/$safeProjectName$.csproj"
```

Leaving issue open to consider doing some basic check that primaryOutput don't start with `/`...

**Comment by ccpu** [comment]
@DavidKarlas thanks for the advice, removing `/` did solve the problem, however, I have discovered a new problem:

To prevent errors related to the internal references, I have to replace the `$safeProjectName$.csproj` to `$safeProjectName$.csproj.xml`; template engine should rename it as configured below:

```
  "sources": [
    {
      "modifiers": [
        {
          "rename": {
            "$safeProjectName$.csproj.xml": "$safeProjectName$.csproj"
          }
        }
      ]
    }
  ],
```

[https://github.com/ccpu/dotnet-template-primaryOutputs-issue/blob/7257bc538873a6f12d4446e598d9575f8e70003d/.template.config/template.json#L22](https://github.com/ccpu/dotnet-template-primaryOutputs-issue/blob/7257bc538873a6f12d4446e598d9575f8e70003d/.template.config/template.json#L22)

Template engine converts the file name correctly, however `dotnet restore` throw the same error as above.

Any suggestion for this issue?

**Comment by DavidKarlas** [comment]
Don't name your `$safeProjectName$.csproj.xml` file with `.xml` at end, remove `.xml` part and no need to do `modifiers` and `rename` if you look at samples at https://github.com/dotnet/dotnet-template-samples they all have .csproj files...

Idea behind template engine is... Your template should be buildable and usable without running it through template engine.... Using template engine(`dotnet new`) will enhance that project that is already usable...
So its much easier to modify template if you can use it in VS without running through  template engine.

**Comment by ccpu** [comment]
Thanks for advice feel free to close issue if you think its not a bug.

**Comment by vlada-shubina** [comment]
> @DavidKarlas David Karlas FTE thanks for the advice, removing `/` did solve the problem, however, I have discovered a new problem:
> 
> To prevent errors related to the internal references, I have to replace the `$safeProjectName$.csproj` to `$safeProjectName$.csproj.xml`; template engine should rename it as configured below:
> 
> ```
>   "sources": [
>     {
>       "modifiers": [
>         {
>           "rename": {
>             "$safeProjectName$.csproj.xml": "$safeProjectName$.csproj"
>           }
>         }
>       ]
>     }
>   ],
> ```
> 
> https://github.com/ccpu/dotnet-template-primaryOutputs-issue/blob/7257bc538873a6f12d4446e598d9575f8e70003d/.template.config/template.json#L22
> 
> Template engine converts the file name correctly, however `dotnet restore` throw the same error as above.
> 
> Any suggestion for this issue?

@ccpu this is a known issue and will be fixed in https://github.com/dotnet/templating/issues/2453. 
I will keep this ticket opened until fix is merged. Thank you.

**Comment by vlada-shubina** [comment]
The issue with incorrect primary outputs in case non-default target or source rename is used is fixed with #2690.
